qiskit/aqua/operators/evolutions/evolved_op.py

```diff
@@ -130,22 +134,24 @@ def assign_parameters(self, param_dict: dict) -> OperatorBase:
             self.primitive.bind_parameters(param_dict), coeff=param_value)  # type: ignore
 
     def eval(self,
-             front: Union[str, dict, np.ndarray,
-                          OperatorBase] = None) -> Union[OperatorBase, float, complex]:
+             front: Optional[Union[str, dict, np.ndarray,
+                                   OperatorBase]] = None) -> Union[OperatorBase, float, complex]:
         return cast(Union[OperatorBase, float, complex], self.to_matrix_op().eval(front=front))
 
     def to_matrix(self, massive: bool = False) -> Union[np.ndarray, List[np.ndarray]]:
         if self.primitive.__class__.__name__ == ListOp.__name__:
-            return \
-                [op.exp_i().to_matrix(massive=massive) *
-                 self.primitive.coeff * self.coeff  # type: ignore
-                 for op in self.primitive.oplist]  # type: ignore
+            return [
+                op.exp_i().to_matrix(massive=massive)
+                * self.primitive.coeff  # type: ignore
+                * self.coeff
+                for op in self.primitive.oplist  # type: ignore
+            ]
 
         prim_mat = -1.j * self.primitive.to_matrix()  # type: ignore
         # pylint: disable=no-member
         return scipy.linalg.expm(prim_mat) * self.coeff
 
-    def to_matrix_op(self, massive: bool = False) -> OperatorBase:
+    def to_matrix_op(self, massive: bool = False) -> Union[MatrixOp, ListOp]:
```

Comment:
I think the reason that this was stated to return OperatorBase is that this is what the method in primitive_op, that this is overriding, states it returns

Reply:
I think return type of `to_matrix()` is Union[MatrixOp, ListOp] for primitive_op. 

However, I found another problem. We cannot use `TYPE_CHECKING` because this is option in sphinx-autodoc-typehints. https://github.com/agronholm/sphinx-autodoc-typehints#options 

I think we should use mix-in instead of inheritance. However the refactoring is not the purpose of this PR. So I revert this change and I will do it when I refactor the code.